src/auth/ha-auth-flow.ts

```diff
@@ -90,6 +91,36 @@ class HaAuthFlow extends litLocalizeLiteMixin(LitElement) {
     }
   }
 
+  private async _postFlow(url: string, postData: any) {
+    try {
+      const response = await fetch(url, {
+        method: "POST",
+        credentials: "same-origin",
+        body: JSON.stringify(postData),
+      });
+
+      const data = await response.json();
+
+      if (response.ok) {
+        // allow auth provider bypass the login form
+        if (data.type === "create_entry") {
+          this._redirect(data.result);
+          return;
+        }
+
+        await this._updateStep(data);
+      } else {
+        this._state = "error";
+        this._errorMessage = data.message;
+      }
+    } catch (err) {
+      // tslint:disable-next-line: no-console
+      console.error("Error updating auth flow", err);
+      this._state = "error";
+      this._errorMessage = this._unknownError();
+    }
+  }
+
```

Comment:
This code was duplicated below. And i needed it once more, so i've extracted it here.